aws/ecs_cluster.go

```
@@ -0,0 +1,132 @@
+			{
+				Key:   aws.String(tagKey),
+				Value: aws.String(tagValue),
+			},
+		},
+	}
+
+	_, err := svc.TagResource(input)
+	if err != nil {
+		return errors.WithStackTrace(err)
+	}
+	return nil
+}
+
+func getAllEcsClustersOlderThan(awsSession *session.Session, region string, excludeAfter time.Time) ([]*string, error) {
+	awsSession, err := session.NewSession(&awsgo.Config{
+		Region: awsgo.String(region),
+	})
+
+	clusterArns, err := getAllEcsClusters(awsSession)
+	if err != nil {
+		logging.Logger.Errorf("Error getting all ECS clusters")
+		return nil, err
+	}
+
+	var filteredEcsClusters []*string
+	for _, clusterArn := range clusterArns {
+
+		firstSeenTime, err := getClusterTag(awsSession, clusterArn, firstSeenTagKey)
+		if err != nil {
+			logging.Logger.Errorf("Error getting the `first_seen` tag for ECS cluster with ARN %s", *clusterArn)
+			return nil, err
+		}
+
+		if firstSeenTime.IsZero() {
+			err := tagEcsCluster(awsSession, clusterArn, firstSeenTagKey, time.Now().UTC().String())
+			if err != nil {
+				logging.Logger.Errorf("Error tagigng the ECS cluster with ARN %s", *clusterArn)
+				return nil, err
+			}
+		}
+
+		if excludeAfter.After(firstSeenTime) {
+			filteredEcsClusters = append(filteredEcsClusters, clusterArn)
+		}
+	}
+
+	return filteredEcsClusters, nil
+}
+
+func nukeEcsClusters(awsSession *session.Session, ecsClusterArns []*string) error {
+	svc := ecs.New(awsSession)
+
+	numNuking := len(ecsClusterArns)
+
+	if numNuking == 0 {
+		logging.Logger.Infof("No ECS clusters to nuke in region %s", *awsSession.Config.Region)
+		return nil
+	}
+
+	logging.Logger.Infof("Deleting %d ECS clusters in region %s", numNuking, *awsSession.Config.Region)
+
+	var nukedEcsClusters []*string
+	for _, clusterArn := range ecsClusterArns {
+		params := &ecs.DeleteClusterInput{
+			Cluster: clusterArn,
+		}
+		_, err := svc.DeleteCluster(params)
+
+		if err != nil {
+			logging.Logger.Errorf("Error, failed to delete cluster with ARN %s", *clusterArn)
```

Do we need to return the error at this point?